sw/device/lib/dif/dif_template.h.tpl

```diff
@@ -46,273 +43,275 @@ extern "C" {
 
 /**
  * Sets whether a particular interrupt is currently enabled or disabled.
  *
- * @param handle A <peripheral> handle.
+ * @param ${handle} A ${periph_lower} handle.
  * @param irq An interrupt type.
  * @param state The new toggle state for the interrupt.
  * @return The result of the operation.
  */
 DIF_WARN_UNUSED_RESULT
-dif_<ip>_result_t dif_<ip>_irq_set_enabled(const dif_<ip>_t *handle,
-                                           dif_<ip>_irq_t irq,
-                                           dif_<ip>_toggle_t state);
+dif_${ip_snake}_result_t dif_${ip_snake}_irq_set_enabled(const dif_${ip_snake}_t *${handle},
+                                                         dif_${ip_snake}_irq_t irq,
+                                                         dif_${ip_snake}_toggle_t state);
 
 /**
  * Forces a particular interrupt, causing it to be serviced as if hardware had
  * asserted it.
  *
- * @param handle A <peripheral> handle.
+ * @param ${handle} A ${periph_lower} handle.
  * @param irq An interrupt type.
  * @return The result of the operation.
  */
 DIF_WARN_UNUSED_RESULT
-dif_<ip>_result_t dif_<ip>_irq_force(const dif_<ip>_t *handle,
-                                     dif_<ip>_irq_t irq);
+dif_${ip_snake}_result_t dif_${ip_snake}_irq_force(const dif_${ip_snake}_t *${handle},
+                                                   dif_${ip_snake}_irq_t irq);
 
 /**
  * Disables all interrupts, optionally snapshotting all toggle state for later
  * restoration.
  *
- * @param handle A <peripheral> handle.
- * @param snapshot Out-param for the snapshot; may be `NULL`.
+ * @param ${handle} A ${periph_lower} handle.
+ * @param[out] snapshot Out-param for the snapshot; may be `NULL`.
  * @return The result of the operation.
  */
 DIF_WARN_UNUSED_RESULT
-dif_<ip>_result_t dif_<ip>_irq_disable_all(const dif_<ip>_t *handle,
-                                           dif_<ip>_irq_snapshot_t *snapshot);
+dif_${ip_snake}_result_t dif_${ip_snake}_irq_disable_all(const dif_${ip_snake}_t *${handle},
+                                                         dif_${ip_snake}_irq_snapshot_t *snapshot);
 
 /**
  * Restores interrupts from the given snapshot.
  *
- * This function can be used with `dif_<ip>_irq_disable_all()` to temporary
+ * This function can be used with `dif_${ip_snake}_irq_disable_all()` to temporary
  * interrupt save-and-restore.
  *
- * @param handle A <peripheral> handle.
+ * @param ${handle} A ${periph_lower} handle.
  * @param snapshot A snapshot to restore from.
  * @return The result of the operation.
  */
 DIF_WARN_UNUSED_RESULT
-dif_<ip>_result_t dif_<ip>_irq_restore_all(const dif_<ip>_t *handle,
-                                           const dif_<ip>_irq_snapshot_t *snapshot);
+dif_${ip_snake}_result_t dif_${ip_snake}_irq_restore_all(const dif_${ip_snake}_t *${handle},
+                                                         const dif_${ip_snake}_irq_snapshot_t *snapshot);
```

Comment:
Could these functions be defined as well in `dif_ip.c.tpl`? These seem very standard across all IPs.

Reply:
I feel like the value is somewhat lower compared to what this header provides. If people think such a thing would be helpful, we can add a C template at a later date.